train.py

```diff
@@ -351,6 +351,9 @@ def loss_fn(pred, labels):
                 with loss_parallel_ctx():
                     pred = model(input_ids)
                     loss = loss_fn(pred, labels)
+                    # pred.shape=(bs, seq_len, vocab_size)
+                    # need to free to before bwd to avoid peaking memory
+                    del pred
```

Comment:
hmmm I wonder could we really del or free this? iiuc the `pred` is part of the autograd graph that needs to be there for backward computation?

Reply:
I believe `pred` is passed into `loss_fn` and `loss` tracks all the autograd ? We did the same trick in xlformer https://github.com/fairinternal/xlformers/blame/001955c08b7a2c5e008d5c503af985f8c80549f2/train.py#L777

or do you prefer some test to verify numerics ?

Reply:
after some thoughts I think this actually make sense to me!

Reply:
the linter installation in CI seems to be broken, since it appears in other PRs as well. For example, this PR changes README https://github.com/pytorch/torchtitan/pull/385) 

thus landing this PR for now